### PR TITLE
[Demo example]Show how to send email with async_simple

### PR DESCRIPTION
**Pull request by qicosmos**
## Why are these changes needed?

Add a demo to Show how to send email with async_simple.

## Related issue number
none

### CODE REVIEW

**Comment by ChuanqiXu9** [comment]
Please resend this one to trigger CI.

**Comment by ChuanqiXu9** [comment]
There are many NFC and formatted change. I prefer to split them in another PR.

**Comment by qicosmos** [comment]
I use clang-format 13.0.1 format the code, and the code format changed, maybe former code format is older.

**Comment by ChuanqiXu9** [comment]
> I use clang-format 13.0.1 format the code, and the code format changed, maybe former code format is older.

We should format the diff only.

Suggest:
> git diff -U0 --no-color --relative HEAD^ | clang-format -p1 -i

**Comment by ChuanqiXu9** [comment]
The other parts could be other PR.

**Comment by ChuanqiXu9** [comment]
Use `git-clang-format  HEAD^` to format diff only.

**Comment by ChuanqiXu9** [comment]
Please resent this one and see if the format CI works.

**Comment by ChuanqiXu9** [comment]
I would close and reopen this PR to run the new CI.

File: demo_example/CMakeLists.txt [hunk 1]
```diff
@@ -32,3 +32,18 @@ target_link_libraries(http_client async_simple)
 
 add_executable(block_http_server http/block_http/block_http_server.cpp)
 target_link_libraries(block_http_server async_simple)
+
+SET(ENABLE_SSL OFF)
```
Comment (ChuanqiXu9) [inline demo_example/CMakeLists.txt:36]: Why this is disabled by default?

**Review by ChuanqiXu9 — APPROVED** [review]
LGTM basically

File: demo_example/CMakeLists.txt (hunk 1 above)
Comment (qicosmos) [reply demo_example/CMakeLists.txt:36]: I think many people don't install openssl and i hope the client can be used immediately, no other dependency as default.

**Pull request merged**